conformance/tests/httproute-service-types.yaml

```diff
@@ -0,0 +1,122 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: service-types
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - matches: [{path: {type: Exact, value: /manual-endpoints}}]
+    backendRefs: [{name: manual-endpoints, port: 8080}]
+  - matches: [{path: {type: Exact, value: /manual-endpointslices}}]
+    backendRefs: [{name: manual-endpointslices, port: 8080}]
+  - matches: [{path: {type: Exact, value: /headless}}]
+    backendRefs: [{name: headless, port: 8080}]
+  - matches: [{path: {type: Exact, value: /headless-manual-endpoints}}]
+    backendRefs: [{name: headless-manual-endpoints, port: 8080}]
+  - matches: [{path: {type: Exact, value: /headless-manual-endpointslices}}]
+    backendRefs: [{name: headless-manual-endpointslices, port: 8080}]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: manual-endpoints
+  namespace: gateway-conformance-infra
+spec:
+  ports:
+  - name: first-port
+    protocol: TCP
+    port: 8080
+    targetPort: 3000
+---
+apiVersion: v1
+kind: Endpoints
+metadata:
+  name: manual-endpoints
+  namespace: gateway-conformance-infra
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: manual-endpointslices
+  namespace: gateway-conformance-infra
+spec:
+  ports:
+  - name: first-port
+    protocol: TCP
+    port: 8080
+    targetPort: 3000
+---
+apiVersion: discovery.k8s.io/v1
+kind: EndpointSlice
+metadata:
+  name: manual-endpointslices
+  namespace: gateway-conformance-infra
+  labels:
+    kubernetes.io/service-name: manual-endpointslices
+addressType: IPv4
+ports:
+- name: first-port
+  port: 3000
+  protocol: TCP
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless
+  namespace: gateway-conformance-infra
+spec:
+  clusterIP: None
+  selector:
+    app: infra-backend-v1
+  ports:
+  - name: first-port
+    protocol: TCP
+    port: 8080
+    targetPort: 3000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: headless-manual-endpoints
+  namespace: gateway-conformance-infra
+spec:
+  clusterIP: None
+  ports:
+  - name: first-port
+    protocol: TCP
+    port: 8080
+    targetPort: 3000
+---
+apiVersion: v1
+kind: Endpoints
+metadata:
+  name: headless-manual-endpoints
+  namespace: gateway-conformance-infra
```

Comment:
In most cases this is really just testing that the EndpointSlice mirroring controller is working. I'd personally prefer skipping this as it's not really something that's testing the GW API implementation and it adds complexity here and in the corresponding go test definition.

Reply:
> In most cases this is really just testing that the EndpointSlice mirroring controller is working.

Can you elaborate?  My intent for testing headless is to ensure that implementations aren't just sending traffic to the cluster IP.

Reply:
Sorry for the confusion, I'm actually referring to any tests that cover custom Endpoints instead of custom EndpointSlices. The EndpointSlice mirroring controller exists to mirror custom Endpoints into custom EndpointSlices so controllers can just read from the EndpointSlice API. That's why I think we should avoid tests that set custom Endpoints because we're really just testing the mirroring controller at that point, not Gateway controllers.

Reply:
Do you think it's worth adding the `endpointslice.kubernetes.io/skip-mirror: true` label to these Endpoints?

Reply:
No, there's no reason for controllers to watch both Endpoints and EndpointSlice APIs. Gateway API is newer than the EndpointSlice API so hopefully all implementations are reading from EndpointSlices, but they definitely shouldn't be reading from both APIs.

Reply:
dropped the `Endpoint` cases